flink-ml-lib/src/test/java/org/apache/flink/ml/clustering/AgglomerativeClusteringTest.java

```diff
@@ -0,0 +1,312 @@
+        verifyMergeInfo(EUCLIDEAN_AVERAGE_MERGE_INFO, outputs[1]);
+
+        // Tests cosine distance with linkage as average.
+        agglomerativeClustering.setDistanceMeasure(CosineDistanceMeasure.NAME);
+        outputs = agglomerativeClustering.transform(inputDataTable);
+        verifyMergeInfo(COSINE_AVERAGE_MERGE_INFO, outputs[1]);
+
+        // Tests manhattan distance with linkage as average.
+        agglomerativeClustering.setDistanceMeasure(ManhattanDistanceMeasure.NAME);
+        outputs = agglomerativeClustering.transform(inputDataTable);
+        verifyMergeInfo(MANHATTAN_AVERAGE_MERGE_INFO, outputs[1]);
+
+        // Tests euclidean distance with linkage as complete.
+        agglomerativeClustering
+                .setDistanceMeasure(EuclideanDistanceMeasure.NAME)
+                .setLinkage(AgglomerativeClusteringParams.LINKAGE_COMPLETE);
+        outputs = agglomerativeClustering.transform(inputDataTable);
+        verifyMergeInfo(EUCLIDEAN_COMPLETE_MERGE_INFO, outputs[1]);
+
+        // Tests euclidean distance with linkage as single.
+        agglomerativeClustering.setLinkage(AgglomerativeClusteringParams.LINKAGE_SINGLE);
+        outputs = agglomerativeClustering.transform(inputDataTable);
+        verifyMergeInfo(EUCLIDEAN_SINGLE_MERGE_INFO, outputs[1]);
+
+        // Tests euclidean distance with linkage as ward.
+        agglomerativeClustering.setLinkage(AgglomerativeClusteringParams.LINKAGE_WARD);
+        outputs = agglomerativeClustering.transform(inputDataTable);
+        verifyMergeInfo(EUCLIDEAN_WARD_MERGE_INFO, outputs[1]);
+
+        // Tests merge info not fully computed.
+        agglomerativeClustering.setComputeFullTree(false);
+        outputs = agglomerativeClustering.transform(inputDataTable);
+        verifyMergeInfo(EUCLIDEAN_WARD_MERGE_INFO.subList(0, 4), outputs[1]);
+    }
+
+    @Test
+    public void testSaveLoadTransform() throws Exception {
+        AgglomerativeClustering agglomerativeClustering =
+                new AgglomerativeClustering()
+                        .setLinkage(AgglomerativeClusteringParams.LINKAGE_AVERAGE)
+                        .setDistanceMeasure(EuclideanDistanceMeasure.NAME)
+                        .setPredictionCol("pred");
+
+        agglomerativeClustering =
+                TestUtils.saveAndReload(
+                        tEnv, agglomerativeClustering, tempFolder.newFolder().getAbsolutePath());
+
+        Table[] outputs = agglomerativeClustering.transform(inputDataTable);
+        verifyClusteringResult(EUCLIDEAN_WARD_K_AS_TWO_RESULT, outputs[0]);
+    }
+
+    @SuppressWarnings("unchecked")
+    private void verifyMergeInfo(List<Row> expected, Table mergeInfoTable) throws Exception {
+        List<Row> mergeInfo =
+                IteratorUtils.toList(tEnv.toDataStream(mergeInfoTable).executeAndCollect());
+        assertEquals(expected.size(), mergeInfo.size());
+        mergeInfo.sort(Comparator.comparingInt(o -> o.getFieldAs(0)));
```

Comment:
`mergeInfo` should be order-sensitive. I think we should verify that the order of the output merge information is also correct.

Reply:
Good catch. I have changed the unit test to verify the order of the merge information.